File: Java/InsertionSort.java
public class InsertionSort{

	public static void main(String[] args) {
		
		int array[] = {9, 1, 8, 2, 7, 3, 6, 5, 4};
        for(int i : array) {
			System.out.print(i + " ");
		}

		insertionSort(array);
        System.out.print("\n");

		for(int i : array) {
			System.out.print(i + " ");
		}
	}

	private static void insertionSort(int[] array) {
		for(int i = 1; i < array.length; i++) { //begins at index 1 because we compare to the left, compare to index 0 at first 
			int temp = array[i];
			int j = i - 1;
			
			while(j >= 0 && array[j] > temp) {
				array[j + 1] = array[j];
				j--;
			}
			array[j + 1] = temp;
		}
	}
}

File: Python/Insertion_Sort.py
array = [8, 7, 9, 2, 3, 1, 5, 4, 6]

print(f"Array to be sorted: {array}\n")

def insertionSort(array):
    for i in range(1, len(array)):
        temp = array[i]
        j = i-1

        while j>=0 and array[j]>temp:
            array[j+1] = array[j]
            j-=1
        array[j+1] = temp 
        print(array)


insertionSort(array)
print(f"Sorted array: {array}\n")